TLM/TLM/Custom/AI/CustomTramBaseAI.cs

```diff
@@ -713,62 +1082,98 @@ public class CustomTramBaseAI : TramBaseAI { // TODO inherit from VehicleAI (in
-                frameData.m_lightIntensity.w = ((turnAngle <= 0.1f || !mayBlink) ? 0f : 5f);
+                frameData.m_lightIntensity.y = springs.z >= -0.1f ? 0.5f : 5f;
+                frameData.m_lightIntensity.z = turnAngle >= -0.1f || !mayBlink ? 0f : 5f;
+                frameData.m_lightIntensity.w = turnAngle <= 0.1f || !mayBlink ? 0f : 5f;
             }
-            frameData.m_underground = ((vehicleData.m_flags & Vehicle.Flags.Underground) != (Vehicle.Flags)0);
-            frameData.m_transition = ((vehicleData.m_flags & Vehicle.Flags.Transition) != (Vehicle.Flags)0);
-            //base.SimulationStep(vehicleID, ref vehicleData, ref frameData, leaderID, ref leaderData, lodPhysics);
+
+            frameData.m_underground = (vehicleData.m_flags & Vehicle.Flags.Underground) != 0;
+            frameData.m_transition = (vehicleData.m_flags & Vehicle.Flags.Transition) != 0;
+
+            // base.SimulationStep(vehicleID, ref vehicleData, ref frameData, leaderID, ref leaderData, lodPhysics);
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void UpdatePathTargetPositions(TramBaseAI tramBaseAI, ushort vehicleID, ref Vehicle vehicleData, Vector3 refPos1, Vector3 refPos2, ushort leaderID, ref Vehicle leaderData, ref int index, int max1, int max2, float minSqrDistanceA, float minSqrDistanceB) {
-            Log.Error($"CustomTramBaseAI.InvokeUpdatePathTargetPositions called! tramBaseAI={tramBaseAI}");
+        [UsedImplicitly]
+        public static void UpdatePathTargetPositions(TramBaseAI tramBaseAi,
+                                                     ushort vehicleId,
+                                                     ref Vehicle vehicleData,
+                                                     Vector3 refPos1,
+                                                     Vector3 refPos2,
+                                                     ushort leaderId,
+                                                     ref Vehicle leaderData,
+                                                     ref int index,
+                                                     int max1,
+                                                     int max2,
+                                                     float minSqrDistanceA,
+                                                     float minSqrDistanceB) {
+            Log._DebugOnlyError($"CustomTramBaseAI.InvokeUpdatePathTargetPositions called! " +
+                                $"tramBaseAI={tramBaseAi}");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static float GetMaxSpeed(ushort leaderID, ref Vehicle leaderData) {
-            Log.Error("CustomTrainAI.GetMaxSpeed called");
+        [UsedImplicitly]
+        private static float GetMaxSpeed(ushort leaderId, ref Vehicle leaderData) {
+            Log._DebugOnlyError("CustomTrainAI.GetMaxSpeed called");
```

Comment:
need to call Error, not _DebugOnlyError

Reply:
in Release the function is not eliminated, the Log call is eliminated